prow/cmd/deck/main.go

```diff
@@ -1004,7 +1005,9 @@ func renderSpyglass(ctx context.Context, sg *spyglass.Spyglass, cfg config.Gette
 		return "", fmt.Errorf("error when resolving real path %s: %v", src, err)
 	}
 	src = realPath
-
+	if err := ValidatePath(cfg, src); err != nil {
+		return "", err
```

Comment:
It looks like this error will still reach the logs and be treated as 5xx instead of being treated as a 4xx, a correct response to an invalid request. Please ensure that we are handling invalid paths as request errors in all uses.

Reply:
I agree 100% that these should be client errors (4xx), but since it requires some restructure/refactor would it be possible to do in a follow-up?